fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -303,10 +308,21 @@ private void pruneTables(OptExpression tree, TaskContext rootTaskContext, Column
     private OptExpression logicalRuleRewrite(ConnectContext connectContext,
                                              OptExpression tree,
                                              TaskContext rootTaskContext) {
+        Boolean isShortCircuit = tree.getShortCircuit();
```

Comment:
you can do all modify before rewriteAndValidatePlan in optimizeByCost, and refactor to a method

Reply:
done